File: UtilitiesVB/Base Functionality Extensions/ToSortedDictionary.cs
using System.Collections.Generic;
using System.Data;
using System.Linq;

namespace UtilitiesVB
{

    public static class ToSortedDictionaryType
    {
        public static SortedDictionary<K, V> ToSortedDictionary<K, V>(this Dictionary<K, V> existing)
        {
            return new SortedDictionary<K, V>(existing);
        }

        public static SortedDictionary<string, bool> SearchSortedDictKeys(SortedDictionary<string, bool> source_dict, string search_string)
        {

            var filtered_cats = (from x in source_dict
                                 where x.Key.Contains(search_string)
                                 select x).ToDictionary(x => x.Key, x => x.Value);
            return new SortedDictionary<string, bool>(filtered_cats);
        }
    }
}

File: UtilitiesVB/Base Functionality Extensions/ToSortedDictionary.vb
Imports System.Runtime.CompilerServices

Public Module ToSortedDictionary
    <Extension()>
    Public Function ToSortedDictionary(Of K, V)(ByVal existing As Dictionary(Of K, V)) As SortedDictionary(Of K, V)
        Return New SortedDictionary(Of K, V)(existing)
    End Function

    Public Function SearchSortedDictKeys(
        source_dict As SortedDictionary(Of String, Boolean),
        search_string As String) _
        As SortedDictionary(Of String, Boolean)

        Dim filtered_cats = (From x In source_dict
                             Where x.Key.Contains(search_string)
                             Select x).ToDictionary(
                             Function(x) x.Key,
                             Function(x) x.Value)
        Return New SortedDictionary(Of String, Boolean)(filtered_cats)
    End Function
End Module
